### PR TITLE
eck-resources chart

### PR DESCRIPTION
**Pull request by iblackman**
First version of the chart

This was being developed in another repo and the commit history can be found in this PR https://github.com/powerhome/APP/pull/1286

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,21 @@
...
+## Depencies
+This charts expects that the [ECK](https://github.com/elastic/cloud-on-k8s) is already installed and configured in the Kube cluster.
+
+## Configuration
+Each variable has a comment to what is can be used for in the [values.yaml](./values.yaml).
+
+Besides that there are some example files in [examples/](./examples/).
+
+## Configure OICD
+This chart allows you to configure login using OICD but just as a reminder, this feature isn't available in the basic license.
```
Comment (benlangfeld) [inline README.md]:
```suggestion
## Configure OIDC
This chart allows you to configure login using OIDC but just as a reminder, this feature isn't available in the basic license.
```

**Comment by benlangfeld** [comment]
This needs a portal.yml, mkdocs.yml and such. @powerhome/guardians-of-the-galaxy can help if you need it, but the docs are at https://portal.powerapp.cloud/docs/default/system/portal/modeling-software/

**Comment by iblackman** [comment]
> This needs a portal.yml, mkdocs.yml and such. @powerhome/guardians-of-the-galaxy can help if you need it, but the docs are at https://portal.powerapp.cloud/docs/default/system/portal/modeling-software/

Why do you think that needs a portal files? This won't be an application. It is a chart that can be used to deploy things.

**Comment by benlangfeld** [comment]
> > This needs a portal.yml, mkdocs.yml and such. @powerhome/guardians-of-the-galaxy can help if you need it, but the docs are at https://portal.powerapp.cloud/docs/default/system/portal/modeling-software/
> 
> Why do you think that needs a portal files? This won't be an application. It is a chart that can be used to deploy things.

Still needs to be modelled in Portal as a library. The systems deployed using this can then model a dependency on it. The docs for this chart would then also be available in Portal. Just because this thing itself doesn't run anywhere continuously, it is still a software asset that needs to be present in our catalog of software assets.

**Comment by iblackman** [comment]
> > > This needs a portal.yml, mkdocs.yml and such. @powerhome/guardians-of-the-galaxy can help if you need it, but the docs are at https://portal.powerapp.cloud/docs/default/system/portal/modeling-software/
> > 
> > 
> > Why do you think that needs a portal files? This won't be an application. It is a chart that can be used to deploy things.
> 
> Still needs to be modelled in Portal as a library. The systems deployed using this can then model a dependency on it. The docs for this chart would then also be available in Portal. Just because this thing itself doesn't run anywhere continuously, it is still a software asset that needs to be present in our catalog of software assets.

ohh Gotcha

File: charts/eck-resources/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# eck-resources-chart
+Helm chart for easily templating ECK resources for deploying Elasticsearch, Kibana etc with reduced duplication.
+
+It allows the user to create some of the custom resources managed by ECK in a simpler way taking advantes of helm templating.
```
Comment (djtaylor) [inline charts/eck-resources/README.md]:
```suggestion
It allows the user to create some of the custom resources managed by ECK in a simpler way by taking advantage of Helm templating.
```

File: charts/eck-resources/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+## Depencies
+This charts expects that the [ECK](https://github.com/elastic/cloud-on-k8s) is already installed and configured in the Kube cluster.
+
+## Configuration
+Each variable has a comment to what is can be used for in the [values.yaml](./values.yaml).
+
+Besides that there are some example files in [examples/](./examples/).
+
+## Configure OIDC
+This chart allows you to configure login using OIDC but just as a reminder, this feature isn't available in the basic license.
```
Comment (djtaylor) [inline charts/eck-resources/README.md]:
```suggestion
This chart allows you to configure login using OIDC. Note that this feature isn't available in the basic license.
```

File: charts/eck-resources/examples/production.yaml [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    volumeClaimTemplates:
+    - metadata:
+        name: elasticsearch-data # Do not change this name unless you set up a volume mount for the data path.
+      spec:
+        accessModes:
+        - ReadWriteOnce
+        resources:
+          requests:
+            storage: 200Gi
+        storageClassName: "staging-performance"
```
Comment (djtaylor) [inline charts/eck-resources/examples/production.yaml]: If this is an example for a production environment, this should probably be:

```suggestion
        storageClassName: "production-performance"
```

File: charts/eck-resources/examples/production.yaml [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    volumeClaimTemplates:
+    - metadata:
+        name: elasticsearch-data # Do not change this name unless you set up a volume mount for the data path.
+      spec:
+        accessModes:
+        - ReadWriteOnce
+        resources:
+          requests:
+            storage: 2Gi
+        storageClassName: "staging-performance"
```
Comment (djtaylor) [inline charts/eck-resources/examples/production.yaml]: Same here, if this is a production example:

```suggestion
        storageClassName: "production-performance"
```

**Review by djtaylor — CHANGES_REQUESTED** [review]

**Comment by djtaylor** [comment]
Since you are using https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack/templates as a reference, I would recommend placing that as a comment somewhere in the README, so we can keep track of the upstream (and possibly make updates based on it or submit PRs).

**Comment by iblackman** [comment]
@benlangfeld I was gonna add guardians to review the portal files but I don't think they have access to this repo.

**Comment by iblackman** [comment]
> Since you are using https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack/templates as a reference, I would recommend placing that as a comment somewhere in the README, so we can keep track of the upstream (and possibly make updates based on it or submit PRs).

@djtaylor  Done. And I fixed the other comments too.

**Review by djtaylor — APPROVED** [review]

File: portal.yaml [hunk 6]
```diff
@@ -0,0 +1,27 @@
+apiVersion: backstage.io/v1alpha1
+kind: System
+metadata:
+  name: helm-charts
+  title: PHRG helm-charts
+  description: Helm charts created by Power Home Remodling Group.
```
Comment (ryanmrodriguez) [inline portal.yaml:6]: Remodeling*

File: portal.yaml [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  name: eck-resources-chart
+  title: Chart eck-resources
+  description: This component is a helm chart that manages ECK resources.
+  annotations:
+    backstage.io/techdocs-ref: dir:.
+spec:
+  type: library
+  owner: watchtower
+  lifecycle: production
+  subcomponentOf: helm-charts
```
Comment (ryanmrodriguez) [inline portal.yaml:27]: Don't need subcomponentOf as this is a root level Component. I think you mean `system: helm-charts`. Also do we need a system for this?

**Review by ryanmrodriguez — COMMENTED** [review]
Filename should be 'portal.yml'

File: portal.yaml (hunk 7 above)
Comment (iblackman) [reply portal.yaml:27]: You tell me, do I need a system?

**Pull request merged**